content/en/docs/reference/using-api/server-side-apply.md

```diff
@@ -224,16 +224,42 @@ merging, see
-| `//+listMapKey` | `x-kubernetes-list-map-keys` | Slice of map keys that uniquely identify entries for example `["port", "protocol"]` | Only applicable when `+listType=map`. A slice of strings whose values in combination must uniquely identify list entries. While there can be multiple keys, `listMapKey` is singular because keys need to be specified individually in the Go type. | 1.16 |
+| `//+listType` | `x-kubernetes-list-type` | `atomic`/`set`/`map` | Applicable to lists. `set` only applies to lists with scalar elements only. `map` applies to lists of nested types only. `atomic` can apply to any list. If configured as `atomic`, the entire list is completely replaced during merge; a single manager manages the list as a whole at any one time. If `set` or `map`, different managers can manage entries separately. | 1.16          |
+| `//+listMapKey` | `x-kubernetes-list-map-keys` | Slice of map keys that uniquely identify entries for example `["port", "protocol"]` | Only applicable when `+listType=map`. A slice of strings whose values in combination must uniquely identify list entries. While there can be multiple keys, `listMapKey` is singular because keys need to be specified individually in the Go type. The key fields must be scalars. | 1.16 |
 | `//+mapType` | `x-kubernetes-map-type` | `atomic`/`granular` | Applicable to maps. `atomic` means that the map can only be entirely replaced by a single manager. `granular` means that the map supports separate managers updating individual fields. | 1.17 |
 | `//+structType` | `x-kubernetes-map-type` | `atomic`/`granular` | Applicable to structs; otherwise same usage and OpenAPI annotation as `//+mapType`.| 1.17 |
 
+If `listType` is missing, a `patchMergeStrategy=merge` marker will be
+converted into a `listType=map` and the corresponding `patchMergeKey`
+marker will be converted into `listMapKey`.
+
+`atomic` is always recursive and these topology markers won't be
+evaluated within an atomic type.
+
+These markers are specified as comments and don't have to be repeated as
+field tags.
+
+#### Compatibility across topology changes
+
+Changing the topology of types that have existing objects will have
+different consequences depending on the type of change.
+
+When the `listType`, `mapType`, or `structType` changes from
+`map`/`set`/`granular` to `atomic`, the whole list, map or struct of
+existing objects will end-up being owned by actors who owned an element
+of these types. This means that any further change to these objects
+would cause a conflict.
+
+When a list, map, or struct goes from atomic to `map`/`set`/`granular`,
+no conflict will be produced when objects with that structure is later
+changed. For that reason, it is not recommended to change a type from
+`atomic` to `map`/`set`/`granular`.
+
```

Comment:
Not necessary for this PR, but this is where I could add something about what happens to field ownership when the topology changes from atomic to granular right?

We basically want to capture two things:
1. The fact that nested fields previously owned by a manager will no longer be owned by that manager (because it only owns the top level).
2. An example (i.e what happens when a map type on a CRD goes from atomic to granular, what fields does a manager own before and after).

I will write all this up in a follow up PR, but just wanted to confirm that this is basically what you were expecting and were you wanted it to live.

Reply:
Yeah, feel free to send a PR against my branch with that change?

Reply:
done https://github.com/apelisse/kubernetes.github.io/pull/1